hazelcast/src/main/java/com/hazelcast/config/MapConfig.java

```diff
@@ -140,6 +140,7 @@ public class MapConfig implements IdentifiedDataSerializable, NamedConfig {
             .setMaxSizePolicy(DEFAULT_MAX_SIZE_POLICY)
             .setSize(DEFAULT_MAX_SIZE);
     private TieredStoreConfig tieredStoreConfig = new TieredStoreConfig();
+    private List<PartitioningAttributeConfig> partitioningAttributeConfigs;
```

Comment:
Why not list of `String`? Why do we need `PartitioningAttributeConfig`?

Reply:
Yeah, I was also thinking about that. But it seems that we generally do that. It gives us extensibility to add more information to each attribute, but I can't foresee anything that we might need. The YAML config actually uses an array, so it's not possible to extend. Perhaps we can use just strings...

Reply:
Extensibility is the main reason, if `partition-strategy` wasn't a single string, I could've added these attribute fields to it directly. In the future there might be a need to support partially nullable keys or for example type coercion. And given that AttributeStrategy is part of the feature still under development, I would say extensibility is more than warranted.

Thanks for pointing out YAML issues, I'll fix that, it was supposed to be an object too.